### PR TITLE
Improve CLI argument help

### PR DESCRIPTION
**Pull request by codebrainz**
Adds context summary, description and argument value descriptions.

### CODE REVIEW

**Comment by elextr** [comment]
LGBI

**Comment by codebrainz** [comment]
@elextr can you test it, just requires inspecting `geany --help` after rebuild/reinstall.

**Comment by elextr** [comment]
As discussed might be nice if the text referred to the context eg "Use FILE as the libvte.so filename" but not essential.

File: src/libmain.c [hunk 1]
```diff
@@ -119,17 +119,17 @@ static gboolean dummy = FALSE;
 /* in alphabetical order of short options */
 static GOptionEntry entries[] =
 {
-	{ "column", 0, 0, G_OPTION_ARG_INT, &cl_options.goto_column, N_("Set initial column number for the first opened file (useful in conjunction with --line)"), NULL },
-	{ "config", 'c', 0, G_OPTION_ARG_FILENAME, &alternate_config, N_("Use an alternate configuration directory"), NULL },
+	{ "column", 0, 0, G_OPTION_ARG_INT, &cl_options.goto_column, N_("Set initial column number for the first opened file (useful in conjunction with --line)"), N_("COLUMN") },
+	{ "config", 'c', 0, G_OPTION_ARG_FILENAME, &alternate_config, N_("Use an alternate configuration directory"), N_("DIR") },
```
Comment (b4n) [inline src/libmain.c]: I would then try and include the placeholder in the description, like *Use alternate configuration directory DIR* or alike.

**Comment by codebrainz** [comment]
> As discussed might be nice if the text referred to the context eg "Use FILE as the libvte.so filename" but not essential.

Done.

File: src/libmain.c [hunk 2]
```diff
@@ -119,17 +119,17 @@ static gboolean dummy = FALSE;
 /* in alphabetical order of short options */
 static GOptionEntry entries[] =
 {
-	{ "column", 0, 0, G_OPTION_ARG_INT, &cl_options.goto_column, N_("Set initial column number for the first opened file (useful in conjunction with --line)"), NULL },
-	{ "config", 'c', 0, G_OPTION_ARG_FILENAME, &alternate_config, N_("Use an alternate configuration directory"), NULL },
+	{ "column", 0, 0, G_OPTION_ARG_INT, &cl_options.goto_column, N_("Set initial column number for the first opened file (useful in conjunction with --line)"), N_("COLUMN") },
```
Comment (b4n) [inline src/libmain.c]: Same as below, maybe *Set initial column number to COLUMN for the first opened file (useful in conjunction with --line)*

File: src/libmain.c [hunk 3]
```diff
@@ -119,17 +119,17 @@ static gboolean dummy = FALSE;
...
+	{ "config", 'c', 0, G_OPTION_ARG_FILENAME, &alternate_config, N_("Use an alternate configuration directory"), N_("DIR") },
 	{ "ft-names", 0, 0, G_OPTION_ARG_NONE, &ft_names, N_("Print internal filetype names"), NULL },
 	{ "generate-tags", 'g', 0, G_OPTION_ARG_NONE, &generate_tags, N_("Generate global tags file (see documentation)"), NULL },
 	{ "no-preprocessing", 'P', 0, G_OPTION_ARG_NONE, &no_preprocessing, N_("Don't preprocess C/C++ files when generating tags file"), NULL },
 #ifdef HAVE_SOCKET
 	{ "new-instance", 'i', 0, G_OPTION_ARG_NONE, &cl_options.new_instance, N_("Don't open files in a running instance, force opening a new instance"), NULL },
-	{ "socket-file", 0, 0, G_OPTION_ARG_FILENAME, &cl_options.socket_filename, N_("Use this socket filename for communication with a running Geany instance"), NULL },
+	{ "socket-file", 0, 0, G_OPTION_ARG_FILENAME, &cl_options.socket_filename, N_("Use this socket filename for communication with a running Geany instance"), N_("FILE") },
```
Comment (b4n) [inline src/libmain.c]: *Use socket filename FILE for communication with a running Geany instance*

File: src/libmain.c [hunk 4]
```diff
@@ -119,17 +119,17 @@ static gboolean dummy = FALSE;
...
 	{ "list-documents", 0, 0, G_OPTION_ARG_NONE, &cl_options.list_documents, N_("Return a list of open documents in a running Geany instance"), NULL },
 #endif
-	{ "line", 'l', 0, G_OPTION_ARG_INT, &cl_options.goto_line, N_("Set initial line number for the first opened file"), NULL },
+	{ "line", 'l', 0, G_OPTION_ARG_INT, &cl_options.goto_line, N_("Set initial line number for the first opened file"), N_("LINE") },
```
Comment (b4n) [inline src/libmain.c]: *Set initial line number to LINE for the first opened file*

File: src/libmain.c [hunk 5]
```diff
@@ -140,7 +140,7 @@ static GOptionEntry entries[] =
 	{ "no-session", 's', G_OPTION_FLAG_REVERSE, G_OPTION_ARG_NONE, &cl_options.load_session, N_("Don't load the previous session's files"), NULL },
 #ifdef HAVE_VTE
 	{ "no-terminal", 't', 0, G_OPTION_ARG_NONE, &no_vte, N_("Don't load terminal support"), NULL },
-	{ "vte-lib", 0, 0, G_OPTION_ARG_FILENAME, &lib_vte, N_("Filename of libvte.so"), NULL },
+	{ "vte-lib", 0, 0, G_OPTION_ARG_FILENAME, &lib_vte, N_("Filename of libvte.so"), N_("FILE") },
```
Comment (b4n) [inline src/libmain.c]: *Use FILE to provide libvte.so* (or something, I'm not too fond of that one right now)

File: src/libmain.c [hunk 6]
```diff
@@ -140,7 +140,7 @@ static GOptionEntry entries[] =
 	{ "no-session", 's', G_OPTION_FLAG_REVERSE, G_OPTION_ARG_NONE, &cl_options.load_session, N_("Don't load the previous session's files"), NULL },
 #ifdef HAVE_VTE
 	{ "no-terminal", 't', 0, G_OPTION_ARG_NONE, &no_vte, N_("Don't load terminal support"), NULL },
-	{ "vte-lib", 0, 0, G_OPTION_ARG_FILENAME, &lib_vte, N_("Filename of libvte.so"), NULL },
+	{ "vte-lib", 0, 0, G_OPTION_ARG_FILENAME, &lib_vte, N_("Use `FILE' as the dynamically-linked VTE library"), N_("FILE") },
```
Comment (b4n) [inline src/libmain.c]: I don't quite like <code>&#96;'</code> quotes, can't we stick to `''` or `""`? (or here I don't even really feel the need for quotes at all, yet I guess it does make sense to have them)

File: src/libmain.c [hunk 7]
```diff
@@ -535,6 +535,8 @@ static void parse_command_line_options(gint *argc, gchar ***argv)
 
 	context = g_option_context_new(_("[FILES...]"));
 
+	g_option_context_set_summary(context, _("A fast and lightweight IDE."));
+	g_option_context_set_description(context, _("Report bugs to `https://github.com/geany/geany/issues'."));
```
Comment (b4n) [inline src/libmain.c]: same here

File: src/libmain.c (hunk 2 above)
Comment (codebrainz) [reply src/libmain.c]: Done.

File: src/libmain.c (hunk 1 above)
Comment (codebrainz) [reply src/libmain.c]: Done.

File: src/libmain.c (hunk 3 above)
Comment (codebrainz) [reply src/libmain.c]: Done.

File: src/libmain.c (hunk 4 above)
Comment (codebrainz) [reply src/libmain.c]: Done.

File: src/libmain.c (hunk 5 above)
Comment (codebrainz) [reply src/libmain.c]: Done.

File: src/libmain.c (hunk 6 above)
Comment (codebrainz) [reply src/libmain.c]: Done.

File: src/libmain.c (hunk 7 above)
Comment (codebrainz) [reply src/libmain.c]: Done.

File: src/libmain.c [hunk 8]
```diff
@@ -119,17 +119,17 @@ static gboolean dummy = FALSE;
 /* in alphabetical order of short options */
 static GOptionEntry entries[] =
 {
-	{ "column", 0, 0, G_OPTION_ARG_INT, &cl_options.goto_column, N_("Set initial column number for the first opened file (useful in conjunction with --line)"), NULL },
-	{ "config", 'c', 0, G_OPTION_ARG_FILENAME, &alternate_config, N_("Use an alternate configuration directory"), NULL },
+	{ "column", 0, 0, G_OPTION_ARG_INT, &cl_options.goto_column, N_("Set initial column number to COLUMN for the first opened file (useful in conjunction with --line)"), N_("COLUMN") },
+	{ "config", 'c', 0, G_OPTION_ARG_FILENAME, &alternate_config, N_("Use an alternate configuration directory DIR"), N_("DIR") },
```
Comment (b4n) [inline src/libmain.c]: `s/an//`

Comment (codebrainz) [reply src/libmain.c]: Done.

**Comment by elextr** [comment]
WFM

**Pull request merged**